packages/tests/unit-tests/test-cases/expected/inferno/spread-props-attribute.tsx

```diff
@@ -46,32 +41,25 @@ export default class Widget extends BaseInfernoComponent<any> {
     };
   }
 
-  get __state_value(): boolean | undefined {
-    const state = this._currentState || this.state;
-    return this.props.value !== undefined ? this.props.value : state.value;
-  }
-  set_value(value: () => boolean | undefined): any {
-    this.setState((state: any) => {
-      this._currentState = state;
-      const newValue = value();
-      this.props.valueChange!(newValue);
-      this._currentState = null;
-      return { value: newValue };
-    });
-  }
+  value?: boolean;
 
   get restAttributes(): RestProps {
```

Comment:
Why it's so complicated? The following code has the same result: 
```
 const { defaultValue, value, valueChange, visible, ...restProps } = { ...this.props  } as any;
 return restProps;
```

Reply:
Value is TwoWay property
When it used uncontrolled, we get this from state
When it used in controlled, we get it from props (highter priority)

If we will take it only from props, then we will ignore uncontrolled mechanism (when `value` was defined through `defaultValue`).
There is 3 way to define option uncontrolled now:
- not define `value`, then the `defaultValue` will be taken
- define `defaultValue` (not used in jQuery earlier, as I understood)
- use `defaultOptionRules` (these options are applied as `defaultXXX` values)

Earlier I tried to get rid of `defaultXXX` in Inferno(jQuery), but this require a lot of changes and research